src/piece_picker.cpp

```diff
@@ -2921,6 +2921,26 @@ namespace libtorrent
 		return true;
 	}
 
+	bool piece_picker::is_piece_ready(int index) const
```

Comment:
I think this function would be simpler (and more efficient) by not calling ``is_piece_finished`` here, but to embed just the interesting case directly in this function. You will likely save a few lookups

Reply:
Will do. Thanks